### PR TITLE
Structured Logging migration: modify server.go part logs of kube-proxy

### PR DESCRIPTION
**Pull request by lunhuijie**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Structured Logging migration: modify server_windows part logs of kube-proxy.
xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md
#### Which issue(s) this PR fixes:
NONE

Fixes #

#### Special notes for your reviewer:
part of kubernetes/enhancements#1602
As it is related to log, change the log here to structured log.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -584,15 +584,15 @@ func serveHealthz(hz healthcheck.ProxierHealthUpdater, errCh chan error) {
...
+			klog.ErrorS(err, "Healthz server failed")
 			if errCh != nil {
 				errCh <- fmt.Errorf("healthz server failed: %v", err)
 				// if in hardfail mode, never retry again
 				blockCh := make(chan error)
 				<-blockCh
 			}
 		} else {
-			klog.Errorf("healthz server returned without error")
+			klog.ErrorS(err, "Healthz server returned without error")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:595]:
```suggestion
			klog.ErrorS(nil, "Healthz server returned without error")
```

File: cmd/kube-proxy/app/server.go [hunk 2]
```diff
@@ -584,15 +584,15 @@ func serveHealthz(hz healthcheck.ProxierHealthUpdater, errCh chan error) {
...
+			klog.ErrorS(err, "Healthz server failed")
 			if errCh != nil {
 				errCh <- fmt.Errorf("healthz server failed: %v", err)
 				// if in hardfail mode, never retry again
 				blockCh := make(chan error)
 				<-blockCh
 			}
 		} else {
-			klog.Errorf("healthz server returned without error")
+			klog.ErrorS(nil, "Healthz server returned without error")
```
Comment (aojea) [inline cmd/kube-proxy/app/server.go:595]: @thockin is this really an error or an Info?

Comment (lunhuijie) [reply cmd/kube-proxy/app/server.go:595]: Should I change it to an Info? or waiting  someone to confirm?

Comment (aojea) [reply cmd/kube-proxy/app/server.go:595]: we need to wait for an approver, so if is not him, other will find the question in the review and answer it

Comment (lunhuijie) [reply cmd/kube-proxy/app/server.go:595]: ok, thanks a lot

Comment (yangjunmyfm192085) [reply cmd/kube-proxy/app/server.go:595]: What is shown here is Info, If want error, Maybe migrate as
klog.ErrorS(fmt.Errorf("Healthz server returned without error"), "")
/cc @serathius, In this case, do we migrate according to info or err?

File: cmd/kube-proxy/app/server.go [hunk 3]
```diff
@@ -217,7 +217,7 @@ func NewOptions() *Options {
 // Complete completes all the required options.
 func (o *Options) Complete() error {
 	if len(o.ConfigFile) == 0 && len(o.WriteConfigTo) == 0 {
-		klog.Warning("WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.")
+		klog.InfoS("WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:220]: Do we need period?

Comment (lunhuijie) [reply cmd/kube-proxy/app/server.go:220]: Maybe period is ok in here. Because there are no para. Do you think so?

Comment (thockin) [reply cmd/kube-proxy/app/server.go:220]: I usually say no trailing periods on logs

File: cmd/kube-proxy/app/server.go [hunk 4]
```diff
@@ -362,7 +362,7 @@ func (o *Options) writeConfigFile() (err error) {
 		return err
 	}
 
-	klog.Infof("Wrote configuration to: %s\n", o.WriteConfigTo)
+	klog.InfoS("Wrote configuration to", "path", o.WriteConfigTo)
```
Comment (thockin) [inline cmd/kube-proxy/app/server.go:365]: nit: "Wrote configuration" - the "to" is not needed

File: cmd/kube-proxy/app/server.go [hunk 5]
```diff
@@ -428,7 +428,7 @@ func (o *Options) loadConfig(data []byte) (*kubeproxyconfig.KubeProxyConfigurati
 		}
 
 		// Continue with the v1alpha1 object that was decoded leniently, but emit a warning.
-		klog.Warningf("using lenient decoding as strict decoding failed: %v", err)
+		klog.InfoS("Using lenient decoding as strict decoding failed", "err", err)
```
Comment (thockin) [inline cmd/kube-proxy/app/server.go:431]: This should be ErrorS I think?

File: cmd/kube-proxy/app/server.go (hunk 2 above)
Comment (thockin) [reply cmd/kube-proxy/app/server.go:595]: I think Error is appropriuate

File: cmd/kube-proxy/app/server.go [hunk 6]
```diff
@@ -640,14 +640,14 @@ func serveMetrics(bindAddress, proxyMode string, enableProfiling bool, errCh cha
 // TODO: At the moment, Run() cannot return a nil error, otherwise it's caller will never exit. Update callers of Run to handle nil errors.
 func (s *ProxyServer) Run() error {
 	// To help debugging, immediately log version
-	klog.Infof("Version: %+v", version.Get())
+	klog.InfoS("Version:", "version", version.Get())
```
Comment (thockin) [inline cmd/kube-proxy/app/server.go:643]: can we find a better message?  Even "Running" is better

File: cmd/kube-proxy/app/server.go [hunk 7]
```diff
@@ -640,14 +640,14 @@ func serveMetrics(bindAddress, proxyMode string, enableProfiling bool, errCh cha
...
 
 	// TODO(vmarmol): Use container config for this.
 	var oomAdjuster *oom.OOMAdjuster
 	if s.OOMScoreAdj != nil {
 		oomAdjuster = oom.NewOOMAdjuster()
 		if err := oomAdjuster.ApplyOOMScoreAdj(0, int(*s.OOMScoreAdj)); err != nil {
-			klog.V(2).Info(err)
+			klog.V(2).InfoS("Error", "err", err)
```
Comment (thockin) [inline cmd/kube-proxy/app/server.go:650]: I think I'd go with Error here, too

File: cmd/kube-proxy/app/server.go (hunk 2 above)
Comment (aojea) [reply cmd/kube-proxy/app/server.go:595]: why a "no error" message is treated as error, is just for learning purposes, is not that we use stderr here or something

**Comment by pacoxu** [comment]
Since #98913 got an LGTM and this PR is dupped with it.
This one get a lgtm as well...

**Comment by lunhuijie** [comment]
> Since #98913 got an LGTM and this PR is part of it.
> This one gets an lgtm as well...

ok, he did it before me
/closed

**Pull request closed without merging**